homeassistant/components/southern_company/sensor.py

```diff
@@ -0,0 +1,305 @@
+        name="Higher projected monthly usage",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="projected_usage_low",
+        name="Lower projected monthly usage",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="projected_bill_amount_low",
+        name="Lower projected monthly cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        key="projected_bill_amount_high",
+        name="Higher projected monthly cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Southern Company sensor."""
+
+    southern_company_connection = hass.data[SOUTHERN_COMPANY_DOMAIN][entry.entry_id]
+
+    coordinator = SouthernCompanyCoordinator(hass, southern_company_connection)
+    entities: list[SouthernCompanySensor] = []
+    await southern_company_connection.get_jwt()
+    for account in await southern_company_connection.get_accounts():
+        device = DeviceInfo(
+            identifiers={((SOUTHERN_COMPANY_DOMAIN), account.number)},
+            name=f"Account {account.number}",
+            manufacturer="Southern Company",
+        )
+        for sensor in SENSORS:
+            entities.append(SouthernCompanySensor(account, coordinator, sensor, device))
+
+    async_add_entities(entities, True)
+
+
+class SouthernCompanySensor(
+    SensorEntity, CoordinatorEntity["SouthernCompanyCoordinator"]
+):
+    """Representation of a Southern company sensor."""
+
+    def __init__(
+        self,
+        account: southern_company_api.Account,
+        coordinator: SouthernCompanyCoordinator,
+        entity_description: SensorEntityDescription,
+        device: DeviceInfo,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._account = account
+
+        self._attr_unique_id = f"{self._account.number}_{self.entity_description.key}"
+        self._attr_name = f"{entity_description.name}"
+        self._attr_device_info = device
+        self._sensor_data = None
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state."""
+        return self._sensor_data
+
+    def _handle_coordinator_update(self) -> None:
+        if self.coordinator.data is not None:
+            self._sensor_data = getattr(
+                self.coordinator.data[self._account.number], self.entity_description.key
+            )
```

Comment:
- This can be directly put into the `native_value` property method.
- Add add a lamba function to the entity description, so we don't need to use `getattr()`.